form-view-model/src/main/kotlin/com/ritense/formviewmodel/web/rest/dto/BusinessRuleError.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.formviewmodel.web.rest.dto
+
+data class BusinessRuleError private constructor (
+    val error: String
+) {
+
+    class Builder {
```

Comment:
I don't see any reason to use Builders with Kotlin. The only use case I see here could be solved by a default value for `error`.

Reply:
@sirolf2009 can you explain why you used this approach?

Reply:
It's to enforce the default "unknown" message. A default value in the constructor doesn't quite work either, because java returns this message as a `String?`, so adding a default value in the constructor will just give an error that `String? cannot be cast to String`. Somewhere, a `?:` check needs to be done and to keep that consistent, I set up this builder.

Reply:
@sirolf2009 

That would work just as well by doing this, right?:
```
class BusinessRuleError (
    error: String?
) {
    val error: String = error ?: UNKNOWN_BUSINESS_RULE_ERROR

    companion object {
        const val UNKNOWN_BUSINESS_RULE_ERROR = "Unknown Business Rule Error"
    }
}
```

This is more Kotlin-like imho. I'll leave it to you to decide.

Reply:
Huh, didn't know you could do that :)

I'll take your proposal, it's a lot less lines of code